include/deal.II/matrix_free/operators.h

```diff
@@ -1896,8 +1916,8 @@ namespace MatrixFreeOperators
           1. / inverse_diagonal_vector.local_element(i);
       }
 
-    inverse_diagonal_vector.update_ghost_values();
-    diagonal_vector.update_ghost_values();
+    diagonal_vector.set_ghost_state(true);
+    inverse_diagonal_vector.set_ghost_state(true);
```

Comment:
Do we really need this? I would simply not have added the call `update_ghost_values`, but left the actual ghost entries in place. If we never call the `update_ghost_entries` (which we probably won't do for pure linear algebra objects as the ones here, see also #2535), one should never see the fact that the vector has ghost entries, except for the very minor jump in memory consumption.

Reply:
I set it to `true` since all ghost values (all zero of them) are present. 

Do we need it? No - all the tests pass without it. We only check that the `src` vector (i.e., `this`) (not the one owned by the diagonal matrix) is ghosted in `LA::d::Vector::scale()` so it doesn't make a difference there.

Since there are no ghost or import indices `update_ghost_values()` does very little, so my final answer is "it doesn't matter but setting it up this way avoids calling `update_ghost_values()` which almost immediately returns".

There may be other ghost update subtleties that I don't know about.

Reply:
Do we need to update the ghost values?

Reply:
No. I've been meaning to get back to this - we should go for the simpler solution implicitly listed in this thread and just not update ghost values (and use the same partitioner).